### DISCUSSION TITLE
Custom Mutation

### DISCUSSION THREAD

**Discussion by ampersandgo** [MichalLytek/typegraphql-prisma#160] [Q&A] [answered]
Is it possible to add a custom mutation resolver?

I want to try adding a custom mutation resolver `signUpUser`. The return type seems to be invalid.

```
import 'reflect-metadata'
import { Arg, Ctx, Mutation, Resolver } from 'type-graphql'
import { User, UserCreateInput } from '../generated/typegraphql-prisma'
import { Context } from '../context'

// custom resolver for User
@Resolver(() => User)
export class CustomUserResolver {
  @Mutation(() => User)
  async signUpUser(
    @Ctx() { prisma }: Context,
    @Arg('data') data: UserCreateInput,
  ): Promise<User> {
    const { firstName, middleName = undefined, lastName = undefined } = data

    try {
      const user = await prisma.user.create({
        data: {
          firstName,
          middleName,
          lastName,
        },
      })

      return user
    } catch (error) {
      throw error
    }
  }
}
```

I always get this error:
![Screen Shot 2021-08-14 at 3 34 32 PM](https://user-images.githubusercontent.com/88924038/129438946-862b4f56-6ba6-4c4f-9f47-4c228f0ac2f0.png)

**Comment by MichalLytek** [comment, marked as answer]
It's your fault, read the error message. `prisma.user.create` argument `data` is missing properties `role` and `contactNumber`.

**Reply by ampersandgo** [reply]
Hi! Actually, I removed it to shorten my code. It's not the problem I meant to share. The TypePropertyDescriptor was my issue and I was able to fix it now using a configuration in typescript. I apologize, this is my first time working with typescript.
